src/wp-includes/global-styles-and-settings.php

```diff
@@ -264,6 +264,23 @@ function ( $item ) {
  * @return bool Returns true if theme or its parent has a theme.json file, false otherwise.
  */
 function wp_theme_has_theme_json() {
+	static $theme_has_support = null;
+
+	/*
+	 * Ignore cache when `WP_DEBUG` is enabled. Why?
+	 * - to avoid interfering with the theme developer's workflow.
+	 * - to reset between each automated test.
+	 *
+	 * @todo Replace `WP_DEBUG` once an "in development mode" check is available in Core.
+	 */
+	if ( WP_DEBUG ) {
+		$theme_has_support = null;
+	}
+
+	if ( null !== $theme_has_support ) {
```

Comment:
See my comment in https://github.com/WordPress/wordpress-develop/pull/3887/files#r1084500610:
```suggestion
	if ( ! WP_DEBUG && null !== $theme_has_support ) {
```

Reply:
Thinking more about the constants, I agree with @azaozz  to _also_ use `WP_RUN_CORE_TESTS` specifically for the tests. In this way, it provides clarity on the purpose of both constants, especially for the future when a "development mode" is introduced into Core.

Reply:
https://github.com/WordPress/wordpress-develop/pull/3887/commits/d0ee04474df46845fa554870a47fc0a861b5705f:

- consolidates the `WP_DEBUG` check as requested
- checks `WP_RUN_CORE_TESTS` per https://github.com/WordPress/wordpress-develop/pull/3887#issuecomment-1401231033 and https://github.com/WordPress/wordpress-develop/pull/3887#discussion_r1085285582

@azaozz and @felixarntz can you re-review please? Thanks!